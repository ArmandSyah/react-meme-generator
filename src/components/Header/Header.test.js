import React from "react";
import { render, cleanup } from "@testing-library/react";

import Header from './Header';

afterEach(cleanup);

it('renders properly', async () => {
    const { getByText } = render(<Header />);

    const memeGeneratorElement = getByText('Meme Generator');

    expect(memeGeneratorElement).toBeVisible();
    expect(memeGeneratorElement).toBeInTheDocument();
    expect(memeGeneratorElement).toHaveClass("header--title");
});

it('renders', async () => {
    const { asFragment } = render(<Header />)
    expect(asFragment()).toMatchSnapshot();
});