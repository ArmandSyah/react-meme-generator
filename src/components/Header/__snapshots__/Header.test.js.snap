// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <img
      class="header--image"
      src="troll-face.png"
    />
    <h2
      class="header--title"
    >
      Meme Generator
    </h2>
    <h4
      class="header--project"
    >
      React Course - Project 3
    </h4>
  </header>
</DocumentFragment>
`;
